coalib/results/TextPosition.py

```diff
@@ -71,11 +70,70 @@ def __ge__(self, other):
         """
         if self.line is None or other.line is None:
             return True
+
         if self.line == other.line:
-            return (True
-                    if self.column is None or
-                    other.column is None
-                    else
-                    self.column >= other.column)
+            if self.column is None or other.column is None:
+                return True
+            else:
+                return self.column >= other.column
         else:
             return self.line > other.line
+
+    def __lt__(self, other):
+        """
+        Test whether ``self`` is behind the other
+        ``TextPosition``.
+
+        If the column in a ``TextPosition`` is ``None``, consider
+        whole line. If the line in a ``TextPosition`` is ``None``,
+        consider whole file.
+
+        :param other: ``TextPosition`` to compare with.
+        :return:      Whether this ``TextPosition`` is behind the other
+                      one.
+        """
+        if self.line is None or other.line is None:
```

Comment:
@aptrishu We have a logically inconsistent situation here:

If any `.line is None`, then `__lt__` returns `False` and `__le__` returns `True`. So the logical consequence is now that `__eq__` must always return `True` in all of those cases. But in fact, `__eq__` only returns `True` in exactly one of those cases: if both `.line` and both `.column` attributes are `None`

@sils @Makman2 @adtac @NiklasMM